lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -50,9 +50,17 @@ defmodule Mix.Tasks.Escript.Install do
   end
 
   def before_install(_src, dst_path) do
-    File.rm(dst_path)
-    File.rm(dst_path <> ".bat")
-    :ok
+    case Mix.Utils.read_path(_src) do
```

Comment:
If we do the above, we should likely swap the order of those two lines: https://github.com/Troush/elixir/blob/check-if-file-is-escript/lib/mix/lib/mix/local/installer.ex#L114-L115